### DISCUSSION TITLE
Disable notifications for Draft Pull Requests

### DISCUSSION THREAD

**Discussion by jg185366** [community/community#10409] [Pull Requests]
Creating a new discussion for this based on an [old thread](https://github.community/t/change-notification-settings-for-draft-pull-requests/2439):

> I would like to only trigger notifications on a Pull Request when it has been marked as “Ready for Review.” Prior to that, I don’t think all watchers on a repository need to be notified that a Draft PR is open.

It's not helpful to me to get notifications when draft pull requests are created or when commits are pushed to them. The PR developer isn't finished yet so I don't want to be notified about this. It would be great if this were at least configurable.

**Comment by mike-odom** [comment]
I'm about to create 10 PRs on a Sunday and would rather my team not get bombarded with all those emails on a weekend. Draft PRs without notifications would be helpful here!

**Comment by cdemyanovich** [comment]
As someone who doesn't like draft pull requests, I would very much like to be able to opt out of notifications, especially email, for them. I'd prefer an account setting, but I'd be happy if the headers and/or content of the email indicated the draft status. At least that would allow me to use my mail client to create a filter.

**Reply by markmcd** [reply]
Huge +1 for adding a header. It's an incremental change that will help a decent number of the folks here.

**Comment by RasmusD** [comment]
Agreed. Being able to opt-out of draft pr notifications would be really great.

**Comment by SamCamhi** [comment]
At the very least DRAFT PR could be labeled as such in the email notification so they can be filtered.

**Comment by mrquincle** [comment]
It would also require the possibility of sending an email when a PR goes from draft mode into non-draft mode, and the other way around.

This would truly fix the draft mode option. There's not much of a difference with non-draft otherwise. 

Having just something like a convention such as `[draft]` or `[WIP]` as a prefix for the name of a regular PR would be more useful to people because they can filter on it in their email clients.

**Comment by cdeutsch** [comment]
My teammates are obsessed with Draft PR's for some reason.

Getting a new notification every time they push a new commit is a big waste of time. 

Really need an option to turn off Draft PR notifications before I just unsubscribe from everything.

**Comment by Olshansk** [comment]
+1

I often use draft PRs to do a "self-review" before I add my colleagues.

I do a check locally as well, but using the github UI forces me to have a different (i.e. external) mindset going into it.

Are there any updates on this?

**Reply by Olshansk** [reply]
**Update**: I spoke to a friend who is also a product manager at GitHub. This is on their radar but not a priority.

**Reply by arxeiss** [reply]
Can you ask him, if there is a way how to put higher priority for it? Like some Feature request pool or something like that?

**Comment by Bikappa** [comment]
Draft PRs are for me the best way to present some non-ready code to colleagues for early reviews.
- the diff viewer is better now with the tree view sidebar
- they can comment the code
- same-window access to checks run compared to a comparison
- some CI tasks might depend on the existence of a PR, so could the related branch deployments.

I see it as a "I'm working on it, please don't bother" mode, with the possibility intentionally propose it to colleagues for opinions or  early catches from knowledgeable people passing by. For this reason I think they should/could remain quiet on the email notifications.

**Comment by jeremycarroll** [comment]
I just had a bot do a mass comment on merged PRs - and these also notified which was not helpful. So the setting should be to notify for PRs in the review phase only

**Comment by haimat** [comment]
Same here - we don't need email notifications on draft pull requests.
Any way to disable these emails?

**Comment by matthewtquinn1** [comment]
I've requested the relevant dev team with Microsoft to introduce this feature. Can people please up vote it? https://developercommunity.visualstudio.com/t/Only-post-published-Pull-Requests-to-Tea/10285656

**Comment by Olshansk** [comment]
I know features are ALWAYS 10x more complex than one imagines, but I feel like this has to be easy enough for an `intern to do in < 2 months (I'm trying to be ultra liberal with timelines)` and would be **real impact to real users**.

@github Pretty please 🥺

**Comment by nutjob4life** [comment]
I attack reviews for pull requests with zeal because I like to keep my notification box empty ("inbox zero"). 

Only too late I see some of the darn things were still in draft! D'oh! I approved a draft!

Thanks, GitHub 😑

**Reply by JonAbrams** [reply]
Why is it even possible review a draft PR? What's the point of having draft PRs if they're identical to normal ones?

**Comment by shaina-aq** [comment]
Yes agreed! Would appreciate if I didn't send my team members unnecessary emails when I commit to a branch I have a draft PR for because it triggers PR checks which are failed.

**Comment by oerp-odoo** [comment]
I also get notifications about failed CI via email. And when I am implementing some changes to CI, I get bombarded about it in an email. Its OK, to get it on normal PRs, but there should be an option to disable notifications per PR. I guess there is no such option? You can either turn off all notifications per repository basis or get all the spam?..

**Comment by robinovitch61** [comment]
Would love for this to be implemented -- would dramatically improve the notifications experience for people like me who like to put up drafts for self-review as well as my colleagues who I don't want to bother for review yet!

**Comment by troglobite** [comment]
I would also love for this to be configurable as I am the only one on my team that uses draft PR's early on in my workflow and my team mates receive the emails, miss the draft labels as they don't really use the feature, and start reviewing code that is not expected to work yet in the context that it is.

I in the blog post that announced the Draft PR that it discusses if you have a CODEOWNERS files it should suppress notifications on draft PR's. We do have a CODEOWNERS file set up in the repo that appears to be setup correctly, but the notification are still being sent out so I'm wondering if this is a bug or perhaps I missed an announcement of this coming back?


- https://github.blog/2019-02-14-introducing-draft-pull-requests/#:~:text=Change%20the%20status%20to%20%E2%80%9CReady%20for%20review%E2%80%9D%20near,until%20it%20is%20marked%20as%20ready%20for%20review

> Also, if you have a [CODEOWNERS file](https://github.blog/2017-07-06-introducing-code-owners/) in your repository, a draft pull request will suppress notifications to those reviewers until it is marked as ready for review.

**Reply by arxeiss** [reply]
If I'm not mistaken, they receive notification not because they are in Code owners. But because you they checked in Watch dropdown they want to receive notifications for All activity or Pull requests.

**Comment by jlebar** [comment]
Or even put **some** indication in the email that it's a draft PR so I don't click on it.

**Comment by ryanleecode** [comment]
This is extremely critical especially because its easier to catch self-made errors reviewing your own PR (in draft) on GH itself, rather than in your text-editor/IDE. i should not have to worry that I'm spamming my teammates with too many emails.

**Comment by MetRonnie** [comment]
Currently there is an option to customise notifications. Would be great to add "Marked as draft" and "Marked as ready for review" to the list.

![image](https://github.com/community/community/assets/61982285/e9cdcc68-adf6-47ce-9d3b-33fb3d23b67b)

**Reply by Olshansk** [reply]
I'm thinking we should start mentioning people here 1 by 1: https://github.com/orgs/github/people

@aashah How can the community help make this happen?

**Reply by ebndev** [reply]
Hi @Olshansk,

Thanks for taking the time to share your feedback and we appreciate your patience! You can check our [public roadmap](https://github.com/github/roadmap) to see the kinds of work we are focusing on for the coming year. Smaller fixes won’t be listed, but are often included when work is happening adjacent to it.

While it has been some time since this thread has been created, the absence of an update is simply that - there has been no update because there is so far nothing to report. That doesn't mean it isn't being worked on or that it won't be considered in the future. 

As a GitHub user, please continue to share your feedback on this Discussion space with the correct tags (include examples), and trust that we are always working to make GitHub as awesome as possible. We can't guarantee that every bit of feedback sees action, but we are paying attention.

**Please note, contacting any member of GitHub Staff via unsolicited mentions or pings is strongly discouraged and may be considered a violation of our prohibition against harassment**. Please review our [Code of Conduct](https://docs.github.com/en/site-policy/github-terms/github-community-code-of-conduct) for more information.

Let's work together to keep GitHub Community a place where people feel safe to participate by being respectful of them and their time.

Any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.

**Reply by ay-azara** [reply]
> While it has been some time since this thread has been created, the absence of an update is simply that - there has been no update because there is so far nothing to report. That doesn't mean it isn't being worked on or that it won't be considered in the future.

@ebndev Sure but in the absence of an update it also means it may never be considered or may be considered 10 years from now. [This request ](https://github.com/orgs/community/discussions/5289) has been sitting here for ages, is highly upvoted and doesn't seem that hard to implement but without knowing what your priorities are, what metrics you all use to determine what requests are worth working on I can't know I'll ever see it.

Apparently we'll see [AI code review](https://github.com/github/roadmap/issues/1022) before we get the ability to keep team members on a review after one of them makes a review. 😕

**Comment by KUGA2** [comment]
This message was added recently:

![image](https://github.com/community/community/assets/10581931/74129e28-2f91-4ed5-a8e0-1ca522fe1a0a)

I think this is very missleading. If the author added reviewers, they are indeed notified! However, it does not add automatic reviewers (from branch policy). The message should say: _"Automatic Reviewers are not added in Draft Mode"_ The current message is just imprecise if not wrong!

**Comment by Olshansk** [comment]
@Github An idea to make it even easier: A label that says `Draft` in the email would help us build appropriate email filters!

**Comment by lassebuur** [comment]
Our team would also highly appreciate the ability to disable notifications from draft PRs!

**Comment by gmelodie** [comment]
We'd also very much appreciate it if you could opt out of draft PR notifications

**Comment by nutjob4life** [comment]
![Mr Bean Waiting meme with four panels featuring Rowan Atkinson as Mister Bean waiting for a bus](https://github.com/user-attachments/assets/80ba986d-3b75-4ee1-9974-b3ae1b7ac8cc)

**Comment by orassyag** [comment]
Late joiner for this, any progress?

**Reply by oerp-odoo** [reply]
see image above:)

**Comment by alex-sc-kao** [comment]
<img width="498" height="281" alt="image" src="https://github.com/user-attachments/assets/2336c327-4d5e-453b-8a0c-c36109144f7a" />
